src/corehost/cli/hostpolicy.cpp

```diff
@@ -32,12 +32,16 @@ enum StatusCode
 //   "true" if path to the CoreCLR dir can be resolved in "clr_path"
 //    parameter. Else, returns "false" with "clr_path" unmodified.
 //
-bool resolve_clr_path(const arguments_t& args, pal::string_t* clr_path)
+bool resolve_clr_path(const arguments_t& args, const pal::string_t& deps_clr_dir, pal::string_t* clr_path)
 {
+    pal::string_t svc_clr = args.dotnet_runtime_servicing;
+    append_path(&svc_clr, _X("runtime"));
+    append_path(&svc_clr, _X("coreclr"));
+
     const pal::string_t* dirs[] = {
-        &args.dotnet_runtime_servicing, // DOTNET_RUNTIME_SERVICING
-        &args.app_dir,                  // APP LOCAL
-        &args.dotnet_home               // DOTNET_HOME
+        &svc_clr,                       // DOTNET_RUNTIME_SERVICING
+        &deps_clr_dir,                  // DEPS BASED CLR
+        &args.app_dir                   // APP LOCAL
```

Comment:
@schellap Why are changing the order of probing? wont this deprioritize app-local lookup? 

CC @davidfowl